src/main/java/flaxbeard/steamcraft/handler/SteamcraftEventHandler.java

```diff
@@ -1740,6 +1882,477 @@ public void rebreath(LivingAttackEvent event) {
+        int x = event.x;
+        int y = event.y;
+        int z = event.z;
+        ItemStack equipped = player.getCurrentEquippedItem();
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (!canDoVoidThings(equipped) || !(tile instanceof IInventory) ||
+          ((IInventory) tile).getSizeInventory() < 1) {
+            return;
+        }
+
+        if (!equipped.hasTagCompound()) {
+            equipped.setTagCompound(new NBTTagCompound());
+        }
+        if (equipped.stackTagCompound.hasKey("voidInventory")) {
+            NBTTagCompound nbt = equipped.stackTagCompound.getCompoundTag("voidInventory");
+            int existingX = nbt.getInteger("x");
+            int existingY = nbt.getInteger("y");
+            int existingZ = nbt.getInteger("z");
+            if (existingX == x && existingY == y && existingZ == z) {
+                equipped.stackTagCompound.removeTag("voidInventory");
+            } else {
+                equipped.stackTagCompound.setTag("voidInventory", setVoidInventoryNBT(nbt, x, y, z));
+            }
+        } else {
+            equipped.stackTagCompound.setTag("voidInventory", setVoidInventoryNBT(new NBTTagCompound(), x, y, z));
+        }
+    }
+
+    @SubscribeEvent
+    public void sendToVoid(BlockEvent.HarvestDropsEvent event) {
+        if (event.harvester == null || event.block == null) {
+            return;
+        }
+
+        EntityPlayer player = event.harvester;
+        World world = event.world;
+        ItemStack equipped = player.getCurrentEquippedItem();
+        if (!canDoVoidThings(equipped) || player.isSneaking()) {
+            return;
+        }
+
+        if (equipped.hasTagCompound() && equipped.stackTagCompound.hasKey("voidInventory")) {
+            NBTTagCompound nbt = equipped.stackTagCompound.getCompoundTag("voidInventory");
+            int invX = nbt.getInteger("x");
+            int invY = nbt.getInteger("y");
+            int invZ = nbt.getInteger("z");
+            TileEntity tile = world.getTileEntity(invX, invY, invZ);
+            if (tile instanceof IInventory) {
+                addToInventory(event.drops, (IInventory) tile);
+            }
+            tile.updateEntity();
+        } else {
+            InventoryEnderChest ender = player.getInventoryEnderChest();
+            addToInventory(event.drops, ender);
+            ender.saveInventoryToNBT();
+        }
+        event.drops.clear();
+    }
+
+    /**
+     * Gets whether the block can be tilled into farmland.
+     * @param world The world
+     * @param block The block to check
+     * @param x The block's X coordinate
+     * @param y The block's Y coordinate
+     * @param z The block's Z coordinate
+     * @return True if it is dirt or grass, else false.
+     */
+    private boolean isFarmable(World world, Block block, int x, int y, int z) {
+        return (block != null && !block.isAir(world, x, y, z) &&
+          (block == Blocks.dirt || block == Blocks.grass));
```

Comment:
Do we need ```!block.isAir(world, x, y, z)``` here? Wouldn't (block == Blocks.dirt || block == Blocks.grass) return false if it was air?